src/missing.py

```diff
@@ -62,25 +69,102 @@ def _find_for_unittest(self, basedir: Optional[str] = None) -> bool:
             path = os.path.normpath(f'{basedir}/{filename}')
             if path in self._exclude:
                 # skip the explicitly excluded path
+                logger.debug('exclude: %r', path)
                 continue
 
             if os.path.isdir(path):
                 if self._find_for_unittest(path):
                     fail = True
             elif self.RE_TEST_PY.match(filename):
-                if self._check_init_py_exists(path):
-                    fail = True
+                if self._is_tracked(path):
+                    logger.debug('check: %r', path)
+                    if self._check_init_py_exists(path):
+                        fail = True
+                else:
+                    logger.debug('untracked: %r', path)
+            else:
+                logger.debug('no_match: %r', path)
 
         if init_run and fail:
             logger.warning('found missing __init__.py for unittest')
 
         return fail
 
+    def _is_tracked(self, path):
+        if self._tracked_files is None:
+            return True
+        return path in self._tracked_files
+
+    def _find_tracked_files(self, mode):
+        if mode == 'obey_gitignore':
+            return self._list_files_obey_gitignore()
+        elif mode == 'staged_only':
+            return self._list_files_staged_only()
+        else:
+            return None
+
+    def _list_files_staged_only(self):
+        # list staged files
+        tracked_files = (
+            subprocess.check_output(
+                ['git', '--no-pager', 'diff', '--name-only', '--cached', '-z'],
+                encoding='utf-8',
+            )
```

Comment:
nit: I prefer not execute with absolute path, not just execute with `git`. But other package[0] seems use `git` too

[0]: https://github.com/gitpython-developers/GitPython/blob/main/git/cmd.py#L236